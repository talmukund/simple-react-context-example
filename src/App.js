import React, { useState, useContext } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Header from './components/header';
import Container from './components/container'
import Chart from './components/charts';
import { Conetxt } from "./context";
import Modal from './components/modal';
import Form from './components/form'

const useStyles = makeStyles((theme) => ({
  paper: {
      padding: theme.spacing(2),
      color: theme.palette.text.secondary,
      minWidth: 600,
      minHeight: 300
  },
}));

function App() {
  const [openModal, setOpenModal] = useState(false);
  const [data, setData] = useState([]);
  const openEditModal = (value) => { 
    setOpenModal(value);
  }
  const value = { data, setData };
  const classes = useStyles();
  return (
    <Conetxt.Provider value={value}>
      <header >
        <Header />
      </header>
      <main>
        <section>
          <Container>
          <Grid item>
                    <Paper className={classes.paper}><Chart /></Paper>
          </Grid>
          <Grid item>
          <Button variant="contained" color="primary" onClick={()=>openEditModal(true)}>
            edit data
            </Button>
          </Grid>
          </Container>
        </section>
        <section>
          <Modal openModal={openModal} setOpenModal={openEditModal}>
            <Form setOpenModal={openEditModal}/>
          </Modal>
          </section>
      </main>
    </Conetxt.Provider>
  );
}

export default App;
