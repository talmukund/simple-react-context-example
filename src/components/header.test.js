import React from 'react';
import { render } from '@testing-library/react';
import App from './header';

test('renders learn react link', () => {
  const Component = render(<App />);
  expect(Component).toMatchSnapshot()
});
