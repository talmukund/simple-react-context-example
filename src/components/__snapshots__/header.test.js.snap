// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          >
            <button
              aria-label="menu"
              class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              />
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <h6
              class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
            >
              Centime
            </h6>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <button
            aria-label="menu"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <h6
            class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
          >
            Centime
          </h6>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
